app/src/main/java/org/oppia/app/player/state/itemviewmodel/FractionInteractionViewModel.kt

```diff
@@ -1,32 +1,88 @@
 package org.oppia.app.player.state.itemviewmodel
 
 import android.content.Context
+import android.text.Editable
+import android.text.TextWatcher
+import androidx.databinding.Bindable
+import androidx.databinding.Observable
+import androidx.databinding.ObservableField
 import org.oppia.app.R
 import org.oppia.app.model.Interaction
 import org.oppia.app.model.InteractionObject
 import org.oppia.app.model.UserAnswer
 import org.oppia.app.parser.StringToFractionParser
+import org.oppia.app.player.state.answerhandling.AnswerErrorCategory
+import org.oppia.app.player.state.answerhandling.InteractionAnswerErrorReceiver
 import org.oppia.app.player.state.answerhandling.InteractionAnswerHandler
 
 /** [ViewModel] for the fraction input interaction. */
 class FractionInteractionViewModel(
-  interaction: Interaction, private val context: Context
+  interaction: Interaction, private val context: Context, private val interactionAnswerErrorReceiver: InteractionAnswerErrorReceiver
 ) : StateItemViewModel(ViewType.FRACTION_INPUT_INTERACTION), InteractionAnswerHandler {
+  private var pendingAnswerError: String? = null
   var answerText: CharSequence = ""
+  var errorMessage = ObservableField<String>("")
   val hintText: CharSequence = deriveHintText(interaction)
+  val stringToFractionParser: StringToFractionParser
+
+  init {
+    stringToFractionParser = StringToFractionParser()
+    var callback: Observable.OnPropertyChangedCallback = object : Observable.OnPropertyChangedCallback() {
```

Comment:
This should be `val` instead of `var`

Reply:
You have marked this as resolve but I am not able to see any changes as per comment also, the comment has not been replied. Therefore marking this conversation as unresolved.